test/GPUActivityAgentTest.cpp

```diff
@@ -0,0 +1,434 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "config.h"
+
+#include <stdlib.h>
+#include <iostream>
+#include <fstream>
+#include <map>
+#include <memory>
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include "geopm_agent.h"
+#include "geopm_hash.h"
+
+#include "Agent.hpp"
+#include "GPUActivityAgent.hpp"
+#include "geopm/Exception.hpp"
+#include "geopm/Helper.hpp"
+#include "geopm/Agg.hpp"
+#include "MockPlatformIO.hpp"
+#include "MockPlatformTopo.hpp"
+#include "geopm/PlatformTopo.hpp"
+#include "geopm_prof.h"
+#include "geopm_test.hpp"
+
+using ::testing::_;
+using ::testing::Invoke;
+using ::testing::Sequence;
+using ::testing::Return;
+using ::testing::AtLeast;
+using geopm::GPUActivityAgent;
+using geopm::PlatformTopo;
+
+class GPUActivityAgentTest : public :: testing :: Test
```

Comment:
Additional tests that could be added for:
- do_send_policy()
- do_send_sample()
- report_host()?
- policy_names()